docs/pages/react/asset/useCreateAsset.en-US.mdx

```diff
@@ -80,19 +94,21 @@ aggregated for simplicity. The `uploadProgress` is a number from 0-1 which repre
 [tus](https://github.com/tus/tus-js-client) - the value is undefined unless an upload is in progress.
 
 ```tsx
-{
-  data?: Asset,
-  uploadProgress?: number,
-  error?: Error,
-  isError: boolean,
-  isIdle: boolean,
-  isLoading: boolean,
-  isSuccess: boolean,
-  status: 'idle' | 'loading' | 'success' | 'error',
-  mutate: (config: CreateAssetArgs, options?: MutateOptions) => void,
-  mutateAsync: (config: CreateAssetArgs, options?: MutateOptions) => Promise<Asset>,
-  variables?: CreateAssetArgs
-}
+[
```

Comment:
Let's change this to be:

```tsx
{
  data?: Asset | Asset[],
  uploadProgress?: number,
  error?: Error,
  isError: boolean,
  isIdle: boolean,
  isLoading: boolean,
  isSuccess: boolean,
  status: 'idle' | 'loading' | 'success' | 'error',
  mutate: (config: CreateAssetArgs, options?: MutateOptions) => void,
  mutateAsync: (config: CreateAssetArgs, options?: MutateOptions) => Promise<Asset | Asset[]>,
  variables?: CreateAssetArgs | CreateAssetArgs[]
}
```

We should be able to infer the return type based on the input type, using conditional types like `type Flatten<T> = T extends any[] ? T[number] : T;`

Reply:
https://www.typescriptlang.org/docs/handbook/2/conditional-types.html